pkg/controller/migration/convert/network.go

```diff
@@ -198,6 +199,42 @@ func handleCalicoCNI(c *components, install *operatorv1.Installation) error {
 		install.Spec.CalicoNetwork.HostPorts = &hp
 	}
 
+	type TuningSpec struct {
+		Sysctl []operatorv1.Sysctl `json:"sysctl,omitempty"`
+		Type   string              `json:"type"`
+	}
+
+	// CNI tuning plugin
+	if pluginData, ok := c.cni.Plugins["tuning"]; ok {
+		// parse JSON data
+		var tuningSpecData TuningSpec
+		if err := json.Unmarshal([]byte(pluginData.Bytes), &tuningSpecData); err != nil {
+
+			return ErrIncompatibleCluster{
+				err:       "error parsing CNI config plugin type 'tuning'",
+				component: ComponentCNIConfig,
+				fix:       "fix CNI config",
+			}
+		}
+
+		// sysctl settings
+		allowedKeys := map[string]bool{
+			"net.ipv4.tcp_keepalive_intvl":  true,
+			"net.ipv4.tcp_keepalive_probes": true,
+			"net.ipv4.tcp_keepalive_time":   true,
+		}
+		sysctlTuning := []operatorv1.Sysctl{}
+		for _, setting := range tuningSpecData.Sysctl {
+			if allowedKeys[setting.Key] {
```

Comment:
This is  the same as the validation code, I think this errors if the allowed Keys are found, which is the opposite of what is expected.